### PR TITLE
Add telemetry for chunk-dedup build artifacts uploads

### PR DESCRIPTION
**Pull request by alex-peck**
Blobstore produces telemetry on how efficient chunk-deduping was. Let's upload that so we can analyze how well the new build artifacts system is working.

Design:
We will take advantage of blobstore's current telemetry system and extend it to also send data to CustomerIntelligence. 
`BlobStoreClientTelemetryTfs` extends `BlobStoreClientTelemetry` and adds a second telemetry sender.
`CustomerIntelligenceTelemetrySender` aggregates telemetry records from uploads and sends them to CustomerIntelligence when `CommitTelemetry` is called.

### CODE REVIEW

File: src/Microsoft.VisualStudio.Services.Agent/Blob/PipelineTelemetryRecord.cs [hunk 1]
```diff
@@ -31,5 +33,33 @@ public abstract class PipelineTelemetryRecord : BlobStoreTelemetryRecord
...
+            Guid taskInstanceId,
+            uint attemptNumber = 1)
+            : base(level, baseAddress, eventNamePrefix, eventNameSuffix, attemptNumber)
+        {
+            PlanId = planId;
+            JobId = jobId;
+            TaskInstanceId = taskInstanceId;
+        }
+
+        protected override void SetMeasuredActionResult<T>(T value)
```
Comment (alex-peck) [inline src/Microsoft.VisualStudio.Services.Agent/Blob/PipelineTelemetryRecord.cs:53]: See `BlobStoreUtils`
This is set during `clientTelemetry.MeasureActionAsync(`

File: src/Agent.Worker/Build/FileContainerServer.cs [hunk 2]
```diff
@@ -204,6 +204,14 @@ private async Task<UploadResult> ParallelUploadAsync(IAsyncCommandContext contex
             _uploadFinished.TrySetResult(0);
             await uploadMonitor;
 
+            // report telemetry
+            if (uploadToBlob)
+            {
+                var planId = Guid.Parse(context.GetVariableValueOrDefault(WellKnownDistributedTaskVariables.PlanId) ?? Guid.Empty.ToString());
```
Comment (jeschu1) [inline src/Agent.Worker/Build/FileContainerServer.cs:210]: Could Guid.Parse throw here?  Should we be doing TryParse?

**Review by jeschu1 — APPROVED** [review]
Let's double check we don't need a TryParse otherwise LGTM

File: src/Agent.Worker/Build/FileContainerServer.cs (hunk 2 above)
Comment (alex-peck) [reply src/Agent.Worker/Build/FileContainerServer.cs:210]: Done

**Pull request merged**